app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -178,6 +180,26 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         }
     }
 
+    @Override
+    public void onViewRecycled(RecyclerView.ViewHolder holder) {
+        super.onViewRecycled(holder);
+        if (holder.getItemViewType() == VIEWTYPE_MESSAGE)
+            markThisMessageAsRead((Message) getItem(holder.getAdapterPosition()));
```

Comment:
This has the potential to cause bugs. views can be inflated & recycled without being shown in order to calculate the layout size / scroll distance.

Let's find a better trigger to mark views as read.